gpg_group_chat/client/client.py

```diff
@@ -1,4 +1,85 @@
+import socket
+import sys
+import threading
+import select
+
 class Client():
 
-    def start(self):
-        print('Hello friend!')
+    def __init__(self):
+        self._socket = None
+        self._connected = False
+        self._nickname = 'anonymous'
+        self._key = 'no-key'
+        
+    def start(self, server_ip = None, server_port = None, nickname = None):
+        if nickname != None:
+            self._nickname = nickname
+        if server_ip == None:
+            print('no server addr')
+            return
+        self._key = 'no-key'
+        print('Hello friend ' + self._nickname)
+        dest = (server_ip, server_port)
+        try:
+            self._socket = socket.create_connection(dest)
+            #self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            #self._socket.connect(dest)
+        except:
+            print('error opening socket')
+            return;
+        success = self.__start_protocol()
+        if(success):
+            self.__handle_messages()
+        else:
+            print('error to stabilish connection')
+
+    def __start_protocol(self):
+        msg = 'hello, i am ' + self._nickname + ',' + self._key
```

Comment:
The protocol implementation it self should be handle by another pull request that is a scope of another issue